src/core/ui.rs

```diff
@@ -328,24 +288,27 @@ impl Recipient<UIUpdate> for UserInterface {
                 Mouse::Moved(position) => {
                     let inverted = if self.settings.invert_y {-1.0} else {1.0};
                     let delta = self.cursor_2d - position;
+                    let yaw_mod = intersection(&self.settings.yaw_modifier_key, &self.input_state.keys_down);
+                    let pan_mod = intersection(&self.settings.pan_modifier_key, &self.input_state.keys_down);
+                    let pitch_mod = intersection(&self.settings.pitch_modifier_key, &self.input_state.keys_down);
 
-                    if self.input_state.yaw_mod || self.input_state.pitch_mod || self.input_state.pan_mod {
-                        if self.input_state.yaw_mod {
+                    if yaw_mod || pitch_mod || pan_mod {
+                        if yaw_mod {
                             Renderer::id() << MoveEye { scene_id: 0, movement: ::monet::Movement::Yaw(
                                 -delta.x * self.settings.rotation_speed * inverted/ 300.0)
                             };
                             
                             self.cursor_2d = position;
                         }
```

Comment:
I think the *_mod if-statements are doing early returns? Shouldn't they be terminated with a semi-colon?